internal/factsengine/gatherers/mountinfo_test.go

```diff
@@ -0,0 +1,212 @@
+package gatherers_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/moby/sys/mountinfo"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+)
+
+type MountInfoTestSuite struct {
+	suite.Suite
+	mockMountParser *mocks.MountParserInterface
+	mockExecutor    *utilsMocks.CommandExecutor
+}
+
+func TestMountInfoTestSuite(t *testing.T) {
+	suite.Run(t, new(MountInfoTestSuite))
+}
+
+func (suite *MountInfoTestSuite) SetupTest() {
+	suite.mockMountParser = new(mocks.MountParserInterface)
+	suite.mockExecutor = new(utilsMocks.CommandExecutor)
+}
+
+func (suite *MountInfoTestSuite) TestMountInfoParsingSuccess() {
+	suite.mockMountParser.On("GetMounts", mock.Anything).Return([]*mountinfo.Info{
+		{
+			Mountpoint: "/sapmnt",
+			Source:     "10.1.1.10:/sapmnt",
+			FSType:     "nfs4",
+			Options:    "rw,relatime",
+		},
+		{
+			Mountpoint: "/hana/data",
+			Source:     "/dev/mapper/vg_hana-lv_data",
+			FSType:     "xfs",
+			Options:    "rw,relatime",
+		},
+		{
+			Mountpoint: "/hana/log",
+			Source:     "/dev/mapper/vg_hana-lv_log",
+			FSType:     "xfs",
+			Options:    "rw,relatime",
+		},
+	}, nil)
+
+	blkidOutput := []byte(`
+DEVNAME=/dev/mapper/vg_hana-lv_data
+UUID=82e3685c-6c2a-432b-bf8c-a71286248fae
+BLOCK_SIZE=4096
+TYPE=xfs
+`)
+
+	blkidOutputNoUUID := []byte(`
+DEVNAME=/dev/mapper/vg_hana-lv_log
+PARTUUID=82e3685c-6c2a-432b-bf8c-a71286248fae
+BLOCK_SIZE=4096
+TYPE=xfs
+`)
+
+	suite.mockExecutor.
+		On("Exec", "blkid", "10.1.1.10:/sapmnt", "-o", "export").
+		Return(nil, fmt.Errorf("blkid error")).
+		On("Exec", "blkid", "/dev/mapper/vg_hana-lv_data", "-o", "export").
+		Return(blkidOutput, nil).
+		On("Exec", "blkid", "/dev/mapper/vg_hana-lv_log", "-o", "export").
+		Return(blkidOutputNoUUID, nil)
+
+	requestedFacts := []entities.FactRequest{
+		{
+			Name:     "not_mounted",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
+			Argument: "/usr/sap",
+		},
+		{
+			Name:     "shared",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
```

Comment:
I've noticed that sometimes we use the same checkid for factrequests in the tests, while other times we put incrementally increased ones and sometimes we don't put it at all. Should we apply some logic when doing so? if on the contrary, it doesn't matter at all for the tests, should we better just remove it?

No need to change anything, just wondering about this.

Reply:
It is true that it doesn't add a lot of real value. This is already tested in other places. At most, if you don't use the `NewFactGatheredWithRequest`, it could fail, as you forgot to pass the check id.
For me it is a good practice, as the `CheckID` is a mandatory field for our contract data